.github/workflows/_release.yml

```diff
@@ -23,60 +29,71 @@ jobs:
       - run: |
           sudo apt-get update
           sudo apt install -y libcurl4-openssl-dev
-      # Linux/Wasm
-      - run: zig build
-      - run: zip -j fastfec-linux-x86_64-${{ inputs.version }}.zip zig-out/bin/fastfec
-      - run: mv zig-out/lib/libfastfec.so libfastfec-linux-x86_64-${{ inputs.version }}.so
-      - run: zig build -Dtarget=aarch64-linux
-      - run: zip -j fastfec-linux-aarch64-${{ inputs.version }}.zip zig-out/bin/fastfec
-      - run: mv zig-out/lib/libfastfec.so libfastfec-linux-aarch64-${{ inputs.version }}.so
-      - run: zig build -Dwasm
-      - run: mv zig-out/lib/fastfec.wasm libfastfec-${{ inputs.version }}.wasm
-      # Mac
-      - run: zig build -Dtarget=x86_64-macos
-      - run: zip -j fastfec-macos-x86_64-${{ inputs.version }}.zip zig-out/bin/fastfec
-      - run: mv zig-out/lib/libfastfec.dylib libfastfec-macos-x86_64-${{ inputs.version }}.dylib
-      - run: zig build -Dtarget=aarch64-macos
-      - run: zip -j fastfec-macos-aarch64-${{ inputs.version }}.zip zig-out/bin/fastfec
-      - run: mv zig-out/lib/libfastfec.dylib libfastfec-macos-aarch64-${{ inputs.version }}.dylib
-      # Windows
-      - run: zig build -Dtarget=x86_64-windows
-      - run: zip -j fastfec-windows-x86_64-${{ inputs.version }}.zip zig-out/bin/fastfec.exe
-      - run: mv zig-out/lib/fastfec.dll libfastfec-windows-x86_64-${{ inputs.version }}.dll
-      - run: zig build -Dtarget=aarch64-windows
-      - run: zip -j fastfec-windows-aarch64-${{ inputs.version }}.zip zig-out/bin/fastfec.exe
-      - run: mv zig-out/lib/fastfec.dll libfastfec-windows-aarch64-${{ inputs.version }}.dll
-      - uses: actions/upload-artifact@v2
+      - name: Build target
+        if: matrix.os != 'wasm'
+        run: zig build -Dtarget=${{ matrix.architecture }}-${{ matrix.os }}
+      - name: Build target
+        if: matrix.os == 'wasm'
+        run: zig build -Dwasm
+      - name: Zip output executable
+        if: matrix.os != 'wasm'
+        run: zip -j fastfec-${{ matrix.os }}-${{ matrix.architecture }}-${{ inputs.version }}.zip zig-out/bin/fastfec
+      - name: Get library output paths
+        uses: kanga333/variable-mapper@master
+        id: library
+        with:
+          key: "${{ matrix.os }}"
+          map: |
+            {
+              "linux": {
+                "ext": "so",
+                "name": "libfastfec"
+              },
+              "mac": {
+                "ext": "dylib",
+                "name": "libfastfec"
+              },
+              "windows": {
+                "ext": "dll",
+                "name": "fastfec"
+              },
+              "wasm": {
+                "ext": "wasm",
+                "name": "fastfec"
+              }
+            }
+          export_to: output
```

Comment:
Ah, neat! Could maybe have done this natively with include? Or did using include for wasm preclude that?

Reply:
I think the matrix doesn't allow you to map additional data per target. If I'm wrong, I'll open a new PR to fix that

Reply:
If I'm reading the [include docs](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations) correctly, it should. But it's a confusing feature and I'm not 100% on everything it does.